### PR TITLE
chore: update (most) deps

### PR DESCRIPTION
**Pull request by montezume**
#### Summary

Updates deps, but leaves the `test-library` and `react` updates for another PR.

* Closes #1014 
* Closes #1013 
* Closes #1012 
* Closes #1011 
* Closes #1010 
* Closes #1009

### CODE REVIEW

File: .eslintrc.yaml [hunk 1]
```diff
@@ -101,3 +101,4 @@ rules:
   'lines-between-class-members': 0
   'react-hooks/rules-of-hooks': 'error'
   'react-hooks/exhaustive-deps': 'warn'
+  'max-classes-per-file': 0
```
Comment (montezume) [inline .eslintrc.yaml]: I guess this was added in the latest version...

Comment (tdeekens) [reply .eslintrc.yaml]: Yeah. I had this in the mc-be too. I just added it to the files having multiple classes so we create more of them. If you don't care about the rule disabling it entirely is sound obviously.

Comment (montezume) [reply .eslintrc.yaml]: Yeah doing it on a case by case basis might be better

File: .eslintrc.yaml [hunk 2]
```diff
@@ -26,6 +26,9 @@ plugins:
   - 'react'
   - 'react-hooks'
 overrides:
+  - files: ['*.story.js']
+    rules:
+      'max-classes-per-file': 0
```
Comment (montezume) [inline .eslintrc.yaml]: I think it makes more sense to add an override for stories, as they had a bunch of these, and we don't really care about them as much as we do our components

**Pull request merged**